common/util.c

```diff
@@ -13,21 +13,28 @@ int wrap(int i, int max) {
 	return ((i % max) + max) % max;
 }
 
-uint32_t parse_color(const char *color) {
+bool parse_color(const char *color, uint32_t *result) {
 	if (color[0] == '#') {
 		++color;
 	}
-
 	int len = strlen(color);
-	if (len != 6 && len != 8) {
-		sway_log(SWAY_DEBUG, "Invalid color %s, defaulting to color 0xFFFFFFFF", color);
-		return 0xFFFFFFFF;
+	char *ptr;
+	uint32_t parsed = strtoul(color, &ptr, 16);
```

Comment:
[`strtoul`](https://en.cppreference.com/w/c/string/byte/strtoul) ignores whitespace before the number, so `parse_color` would accept "`# fffff`" (with one space) as a valid color. "`#0x0000`" also passes.

Reply:
Thanks. `strtoul` was already being used so I didn't think about it. I changed it so it verifies the first two characters (excluding the `#`) are hex digits. That will catch signs, leading whitespace, and `0x`. Anything further into the string will be caught by the `*ptr != '\0'`.